### PR TITLE
planning: open space, include and visualize partitioned trajectories

### PR DESCRIPTION
**Pull request by deidaraho**

### CODE REVIEW

File: modules/planning/common/open_space_info.cc [hunk 1]
```diff
@@ -22,6 +22,43 @@
 
 namespace apollo {
 namespace planning {
-// TODO(deidaraho): record more trajectory information to info debug
+  void CopyTrajectory(
```
Comment (xiaoxq) [inline modules/planning/common/open_space_info.cc]: Align

File: modules/planning/on_lane_planning.cc [hunk 2]
```diff
@@ -412,11 +412,12 @@ Status OnLanePlanning::Plan(
...
     if (FLAGS_enable_record_debug) {
-      ptr_debug->MergeFrom(frame_->open_space_info().debug_instance());
+      // ptr_debug->MergeFrom(frame_->open_space_info().debug_instance());
+      frame_->mutable_open_space_info()->RecordDebug(ptr_debug);
       ADEBUG << "Open space debug information added!";
       // call open space info load debug
       // TODO(Runxin): create a new flag to enable openspace chart
-      ExportOpenSpaceChart(frame_->open_space_info().debug_instance(),
+      ExportOpenSpaceChart(trajectory_pb->debug(),
                            ptr_debug);
```
Comment (xiaoxq) [inline modules/planning/on_lane_planning.cc]: Fit in one line.

File: modules/planning/common/open_space_info.cc (hunk 1 above)
Comment (deidaraho) [reply modules/planning/common/open_space_info.cc]: > Align

fixed

File: modules/planning/common/open_space_info.cc [hunk 3]
```diff
@@ -22,6 +22,43 @@
 
 namespace apollo {
 namespace planning {
-// TODO(deidaraho): record more trajectory information to info debug
+void CopyTrajectory(
+    const apollo::planning::DiscretizedTrajectory trajectory_src,
+    apollo::common::Trajectory* trajectory_tgt_ptr) {
+    size_t horizon = trajectory_src.NumOfPoints();
```
Comment (xiaoxq) [inline modules/planning/common/open_space_info.cc]: Also fix indents.

File: modules/planning/common/open_space_info.cc [hunk 4]
```diff
@@ -22,6 +22,43 @@
...
+void OpenSpaceInfo::RecordDebug(apollo::planning_internal::Debug* ptr_debug) {
+  // 1, merge debug info into ptr_debug
+  ptr_debug->MergeFrom(debug_instance_);
+
+  // 2, record partitioned trajectories into debug_ptr
+  auto* ptr_partitioned_trajectories = ptr_debug->mutable_planning_data()
+      ->mutable_open_space()->mutable_partitioned_trajectories();
+
+  for (auto iter : paritioned_trajectories_) {
+    auto picked_trajectory = iter.first;
```
Comment (xiaoxq) [inline modules/planning/common/open_space_info.cc]: auto& ?

File: modules/planning/on_lane_planning.cc [hunk 5]
```diff
@@ -686,5 +687,64 @@ void OnLanePlanning::AddOpenSpaceOptimizerResult(
...
+  // if open space info provider success run
+  if (!frame_->open_space_info().open_space_provider_success()) {
+    return;
+  }
+  // if empty return
+  auto chart = debug_chart->mutable_planning_data()->add_chart();
+  auto open_space_debug = debug_info.planning_data().open_space();
+
+  if (open_space_debug.chosen_trajectory().trajectory().empty() ||
+      open_space_debug.chosen_trajectory().trajectory()[0].
```
Comment (xiaoxq) [inline modules/planning/on_lane_planning.cc]: Save `open_space_debug.chosen_trajectory().trajectory()` to variable to shorten code. DRY.

File: modules/planning/common/open_space_info.cc [hunk 6]
```diff
@@ -22,6 +22,43 @@
 
 namespace apollo {
 namespace planning {
-// TODO(deidaraho): record more trajectory information to info debug
+void CopyTrajectory(
+  const apollo::planning::DiscretizedTrajectory trajectory_src,
+  apollo::common::Trajectory* trajectory_tgt_ptr) {
```
Comment (xiaoxq) [inline modules/planning/common/open_space_info.cc]: 4 spaces indents for arguments, 2 for enclosed statements.

Comment (xiaoxq) [reply modules/planning/common/open_space_info.cc]: Try `scripts/clang-format.sh` to train yourself :)

**Comment by Capri2014** [comment]
Hi, I checkout this PR and created a new one to fix the lint issue

**Comment by Capri2014** [comment]
https://github.com/ApolloAuto/apollo/pull/7503

**Pull request closed without merging**